### PR TITLE
Membership checking via NationBuilder API

### LINKED ISSUE #189: Connect Membership controller to NationBuilder
Instead of reading hard-coded emails from a config variable, we should use the NB API to check for membership status. The user needs to have the `national_member` tag to be given access. Let's also preserve a way to hard-code in some extra email addresses, either via the config variable or a list of emails in Auth0.

### PR DESCRIPTION
**Pull request by kyleschmolze**
Closes #189 

Now the Memberships controller returns true if the email is in the whitelist or in the NationBuilder database with the tag `national_member`. Also, I've added a very simple API key check, where we have to set an ENV variable called `API_KEY`, and any service checking the membership endpoint must include that API key as an `api_key` parameter in their request. We can just randomly generate an API key and keep it out of the repo, but include it in the request going from Auth0 to our server.

Before merging this, we should probably close the bug #190 which allows anyone to self-tag as a `national_member`. (edit: merged!)

### CODE REVIEW

**Comment by kyleschmolze** [comment]
When you say "log failure", do you just mean to spit out in the server logs what IPs we're receiving these requests from, but still allow them all through? That sounds smart to me, and we might as well remove the IP whitelist requirement, since we're using the API key now.

**Comment by tilgovi** [comment]
Exactly that. 👍

**Comment by kyleschmolze** [comment]
Ok, done. Look good @tilgovi?

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -9,6 +9,12 @@ def require_nationbuilder_slug
     end
   end
 
+  def require_api_key
+    unless params[:api_key].present? and ENV['API_KEY'].present? and params[:api_key] == ENV['API_KEY']
```
Comment (eventualbuddha) [inline app/controllers/application_controller.rb]: This comparison via `String#==` is vulnerable to a [timing attack](https://en.wikipedia.org/wiki/Timing_attack). Comparing secrets should always be done with a method that doesn't leak information such as [`ActiveSupport::SecurityUtils.secure_compare`](https://api.rubyonrails.org/classes/ActiveSupport/SecurityUtils.html#method-c-secure_compare):

```rb
ActiveSupport::SecurityUtils.secure_compare(params[:api_key], ENV['API_KEY'])
```

Comment (tilgovi) [reply app/controllers/application_controller.rb]: Good catch.

Comment (kyleschmolze) [reply app/controllers/application_controller.rb]: Truth!

**Pull request merged**